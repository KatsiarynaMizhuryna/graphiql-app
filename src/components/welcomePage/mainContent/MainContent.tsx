import { LinkButton } from '@/ui/linkButton';

interface MainContentProps {
  userName: string;
}

const userName = 'John Doy';

export const MainContent = ({ userName }: MainContentProps) => {
  return (<div className="flex-grow container mx-auto flex flex-col items-center justify-between gap-12 sm:gap-16 md:gap-20 lg:gap-24 xl:gap-28">
      <div className="ml-auto">
        <LinkButton className="ml-auto" href="/">
          Main page
        </LinkButton>
      </div>
      <div className="flex-grow place-content-center">
        Welcome Back, {userName}!
      </div>
      <div className="flex flex-row justify-between">
        <LinkButton href="/#">REST Client</LinkButton>
        <LinkButton href="/#">GraphiQL Client</LinkButton>
        <LinkButton href="/#">History</LinkButton>
      </div>
    </div>
  );
};
