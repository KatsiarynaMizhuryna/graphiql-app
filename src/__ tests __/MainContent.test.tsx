import React, { ReactNode } from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MainContent } from '@/components/welcomePage/mainContent/MainContent';
import Link from 'next/link';


interface LinkButtonProps {
  children: ReactNode;
  href: string;
}

jest.mock('@/ui/LinkButton', () => ({
  LinkButton: ({ children, href }: LinkButtonProps) => (
    <Link href={href}>{children}</Link>
  )
}));

jest.mock('@/ui/fonts', () => ({
  Nunito: () => ({
    className: 'nunito',
  }),
}));

jest.mock('@/ui/fonts', () => ({
  oswald: { className: 'oswald-font' },
}));

describe('MainContent component', () => {
  it('renders the welcome message with username', () => {
    const userName = 'John Doe';

    render(<MainContent userName={userName} />);

    expect(screen.getByText(`Welcome Back, ${userName}!`)).toBeInTheDocument();
  });

  it('renders main page link', () => {
    render(<MainContent userName="John Doe" />);

    expect(screen.getByText('Main page')).toBeInTheDocument();
    expect(screen.getByText('Main page').closest('a')).toHaveAttribute('href', '/');
  });

  it('renders REST Client, GraphiQL Client, and History links', () => {
    render(<MainContent userName="John Doe" />);

    expect(screen.getByText('REST Client')).toBeInTheDocument();
    expect(screen.getByText('REST Client').closest('a')).toHaveAttribute('href', '/#');

    expect(screen.getByText('GraphiQL Client')).toBeInTheDocument();
    expect(screen.getByText('GraphiQL Client').closest('a')).toHaveAttribute('href', '/#');

    expect(screen.getByText('History')).toBeInTheDocument();
    expect(screen.getByText('History').closest('a')).toHaveAttribute('href', '/#');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<MainContent userName="John Doe" />);
    expect(asFragment()).toMatchSnapshot();
  });
});