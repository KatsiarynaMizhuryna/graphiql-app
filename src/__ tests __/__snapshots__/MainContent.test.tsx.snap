// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ml-auto place-content-end"
  >
    <a
      href="/"
    >
      Main page
    </a>
  </div>
  <h1
    class="oswald-font text-4xl font-bold mb-10 text-center"
  >
    Welcome Back, John Doe!
  </h1>
  <div
    class="w-full flex justify-between max-sm:flex-col max-sm:text-center max-sm:gap-5 "
  >
    <a
      href="/#"
    >
      REST Client
    </a>
    <a
      href="/#"
    >
      GraphiQL Client
    </a>
    <a
      href="/#"
    >
      History
    </a>
  </div>
</DocumentFragment>
`;
